mantidproject/mantid
# Add reflectometry GUI runs presenter tests

This PR adds missing unit tests for the Runs tab on the ISIS Reflectometry GUI. This includes the following main changes:

- Add a test to ensure the table presenter is updated when doing a transfer of runs from the search box.
- Add unit tests for the live data monitor and a `MockAlgorithmRunner` to support them.
- Generalise some supporting functions in `ParseKeyValueString` to help with converting between strings and maps when comparing results.
- Add unit tests for the new `ParseKeyValueString` behaviour and also split up the large existing tests for these functions into smaller units.

**To test:**

- Code review
- Ensure the parsing still works in the ISIS Reflectometry interface. This is used for the Options column and the stitch parameters:
  - Enter instrument `INTER`, run number `13460`, angle `0.5` and Options `ProcessingInstructions=4` on the Runs tab.
  - On the Experiment tab, enter the output stitch params as `Params=-0.02`
  - Back on the Runs tab, click Process.
- Ensure the live data monitor button still works on the interface:
  - Click Start Monitor and check that it attempts to start the live data monitor (it may not succeed if the instrument isn't currently running but should give a sensible error).

Refs #26512 	

*This does not require release notes* because **it concerns testing only**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
